src/openapi/peridio-device-openapi.yaml

```diff
@@ -272,9 +292,82 @@ paths:
+        </a>
+
+        If successful, the tunnel's state will be `open`. Only tunnels with a state of `requested` can be configured.
+      tags:
+        - "Tunnels"
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              properties:
+                tunnel_prn:
+                  $ref: "peridio-admin-openapi.yaml#/components/schemas/tunnel-prn"
+                cidr_blocks:
+                  $ref: "#/components/schemas/cidrs"
+                  examples: [[10.0.0.1/16, 172.154.0.1/16]]
+                  description: |
+                    A set of CIDR blocks that the device has available for use with remote access. The device's wireguard peer IP as well as wireguard interface IP will be choosen from this set so it must contain at least two distinct IPs.
+                port_ranges:
+                  $ref: "#/components/schemas/port_ranges"
+                  examples: [[45000-46000, 47000-48000]]
+                  description: |
+                    A set of port ranges that the device has available for use with remote access. The device can use this to constrain the pool of ports from which Peridio will choose the wireguard server listen port, that is, the port on which the server is listening for wireguard UDP traffic.
+                device_proxy_port:
+                  type: number
+                  description: The port on which the device is listening for wireguard UDP traffic.
+                  examples: [530000]
+                device_tunnel_port:
+                  type: number
+                  description: The port to which wireguard will forward traffic to on the device.
+                  examples: [22]
+                device_public_key:
+                  $ref: "#/components/schemas/base64"
+                  examples: [Y4nCIXdpb+f3WgPi0377FDPCAfP+st82s98lTRepHEk=]
+                  description: The device's public key for wireguard.
+              required:
+                - cidr_blocks
+                - device_proxy_port
+                - device_public_key
+                - device_tunnel_port
+                - port_ranges
+                - tunnel_prn
+
+      responses:
+        "200":
+          description: Ok.
+          content:
+            application/json:
+              schema:
+                properties:
+                  data:
+                    $ref: "peridio-admin-openapi.yaml#/components/schemas/tunnel"
 
 components:
   securitySchemes:
     MutualTLS:
       type: mutualTLS
       description: The provided client certificate must be signed by a [CA certificate](https://docs.peridio.com/reference/ca-certificates) registered with Peridio.
+  schemas:
+    cidrs:
+      type: array
+      items:
+        type: string
+    port_ranges:
```

Comment:
Hyphen not underscore for schemas. Just to be consistent with how Admin API is written.

`port-ranges:`